src/lib/util/opresolv.cpp

```diff
@@ -410,51 +388,56 @@ static const struct option_resolution_entry *option_resolution_lookup_entry(opti
 }
 
 
+// -------------------------------------------------
+//	lookup_int
+// -------------------------------------------------
 
-int option_resolution_lookup_int(option_resolution *resolution, int option_char)
+int option_resolution::lookup_int(int option_char) const
 {
-	const struct option_resolution_entry *entry;
-	entry = option_resolution_lookup_entry(resolution, option_char);
-	return entry ? entry->u.int_value : -1;
+	auto entry = lookup_entry(option_char);
+	return entry ? entry->int_value() : -1;
 }
 
 
+// -------------------------------------------------
+//	lookup_string
+// -------------------------------------------------
 
-const char *option_resolution_lookup_string(option_resolution *resolution, int option_char)
+const char *option_resolution::lookup_string(int option_char) const
 {
-	const struct option_resolution_entry *entry;
-	entry = option_resolution_lookup_entry(resolution, option_char);
-	return entry ? entry->u.str_value : nullptr;
+	auto entry = lookup_entry(option_char);
+	return entry ? entry->value.c_str() : nullptr;
 }
 
 
+// -------------------------------------------------
+//	find_option
+// -------------------------------------------------
 
-const char *option_resolution_specification(option_resolution *resolution)
+const option_guide *option_resolution::find_option(int option_char) const
 {
-	return resolution->specification;
-}
-
-
-
-const option_guide *option_resolution_find_option(option_resolution *resolution, int option_char)
-{
-	const struct option_resolution_entry *entry;
-	entry = option_resolution_lookup_entry(resolution, option_char);
+	auto entry = lookup_entry(option_char);
 	return entry ? entry->guide_entry : nullptr;
 }
 
 
+// -------------------------------------------------
+//	index_option
+// -------------------------------------------------
 
-const option_guide *option_resolution_index_option(option_resolution *resolution, int indx)
+const option_guide *option_resolution::index_option(int indx) const
 {
-	if ((indx < 0) || (indx >= resolution->option_count))
+	if ((indx < 0) || (indx >= m_entries.size()))
 		return nullptr;
-	return resolution->entries[indx].guide_entry;
+	return m_entries[indx].guide_entry;
 }
 
 
+// -------------------------------------------------
+//	count_options
+// -------------------------------------------------
 
-int option_resolution_countoptions(const option_guide *guide, const char *specification)
+int option_resolution::count_options(const option_guide *guide, const char *specification)
```

Comment:
this should probably return an unsigned integer or a `std::size_t` rather than a signed integer since a negative result would be meaningless